packages/nodejs/test/e2e/remote-comms.test.ts

```diff
@@ -798,4 +798,102 @@ describe.sequential('Remote Communications E2E', () => {
+          maxRetryAttempts: 1, // Only 1 retry attempt before giving up
+        });
+        await kernel2.initRemoteComms({ relays: testRelays });
+
+        // Set up Alice and Bob manually (can't use setupAliceAndBob as it reinitializes comms)
+        const aliceConfig = makeRemoteVatConfig('Alice');
+        const bobConfig = makeRemoteVatConfig('Bob');
+
+        await launchVatAndGetURL(kernel1, aliceConfig);
+        const bobURL = await launchVatAndGetURL(kernel2, bobConfig);
+
+        const aliceRef = getVatRootRef(kernel1, kernelStore1, 'Alice');
+
+        // Establish connection first by sending a successful message
+        await sendRemoteMessage(kernel1, aliceRef, bobURL, 'hello', ['Alice']);
+
+        // Now stop kernel2 to trigger connection loss
+        await kernel2.stop();
+
+        // Wait for connection loss to be detected and reconnection attempts to fail
+        await wait(2000);
+
+        // Send a message that will trigger promise creation and eventual rejection
+        // The message will create a promise with the remote as decider (from URL redemption)
+        // When we give up on the remote, that promise should be rejected
+        // The vat should then propagate that rejection to the promise returned here
+        const messagePromise = kernel1.queueMessage(
+          aliceRef,
+          'sendRemoteMessage',
+          [bobURL, 'hello', ['Alice']],
+        );
+
+        const result = await messagePromise;
+        const response = kunser(result);
+        expect(response).toBeInstanceOf(Error);
+        expect((response as Error).message).toContain(
+          'max retries reached or non-retryable error',
+        );
+      },
+      NETWORK_TIMEOUT * 2,
+    );
+
+    it(
+      'resolves promise after reconnection when retries have not been exhausted',
+      async () => {
+        const { aliceRef, bobURL } = await setupAliceAndBob(
+          kernel1,
+          kernel2,
+          kernelStore1,
+          kernelStore2,
+          testRelays,
+        );
+
+        // Send a message that creates a promise with remote as decider
+        const messagePromise = kernel1.queueMessage(
+          aliceRef,
+          'sendRemoteMessage',
+          [bobURL, 'hello', ['Alice']],
+        );
+
+        // Stop kernel2 before it can respond
+        await kernel2.stop();
+
+        // Wait a bit for connection loss to be detected
+        await wait(500);
+
+        // Restart kernel2 quickly (before max retries, since default is infinite)
+        // The promise should remain unresolved and resolve normally after reconnection
+        const bobConfig = makeRemoteVatConfig('Bob');
+        // eslint-disable-next-line require-atomic-updates
+        kernel2 = (
+          await restartKernelAndReloadVat(
+            kernelDatabase2,
+            false,
+            testRelays,
+            bobConfig,
+          )
+        ).kernel;
+
+        // Wait for reconnection
+        await wait(2000);
```

Comment:
nit: We have both `wait()` and `delay()`? I guess I prefer "wait". Could do that separately.